feat(theme): add initial files to support Excursion theme #153

jason-capsule42 opened the pull request:
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #153

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [ ] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)


## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

jason-capsule42 committed:
feat(theme): add initial files to support Excursion theme #153

jason-capsule42 committed:
perf(deprecated): clean up old deprecated and deprecate newly outdated tokens

jason-capsule42 committed:
feat(classic): add tier 1 classic token structure

jordanjones243 committed:
feat(background): add new background tier 2 tokens

jordanjones243 commented on src/color/background.json:
```diff
@@ -1,4 +1,280 @@
+    "secondary": {
+      "100": {
+        "default": {
+          "value": "{color.base.white.value}",
```
On the figma, I see a token called `--ds-color-base-white`, but not seeing the `100`.

jason-capsule42 committed:
refactor(tokens): update brand tokens with correct values

jason-capsule42 committed:
refactor(tokens): cleanup the tier 1 tokens

jason-capsule42 committed:
fix(grey): correct typo in grey name

jason-capsule42 committed:
feat(container): add values for container tokens

jason-capsule42 committed:
refactor(background): use correct values for white

jason-capsule42 committed:
feat(border): add new tier 2 border tokens

jason-capsule42 committed:
feat(text): add new tier 2 text tokens

jason-capsule42 committed:
feat(text): add new tier 2 Tier tokens

jordanjones243 committed:
feat(icon): add new tier 2 icon tokens

jason-capsule42 committed:
refactor(tokens): update tokens to correct hex codes

jason-capsule42 committed:
feat(theme): add base files and config for jetstream theme

jason-capsule42 committed:
refactor(jetstream): temporarily disable the jetstream theme in the build

jordanjones243 committed:
fix(classic): update tier 1 tokens to correct values

jason-capsule42 committed:
docs(readme): add excurion theme to the resources list

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/AlaskaAirlines/AuroDesignTokens?pullRequest=156) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/AlaskaAirlines/AuroDesignTokens?pullRequest=156) before we can accept your contribution.<br/>**3** out of **4** committers have signed the CLA.<br/><br/>:white_check_mark: jason-capsule42<br/>:white_check_mark: jordanjones243<br/>:white_check_mark: blackfalcon<br/>:x: semantic-release-bot<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/AlaskaAirlines/AuroDesignTokens?pullRequest=156) it.</sub>

blackfalcon commented on scripts/config-excursion.json:
```diff
@@ -0,0 +1,28 @@
+{
+  "include": ["./src/themes/excursion/color/*.json"],
```
```suggestion
```



Using `include` and `source` is not necessary. 

`include` is optional by definition of the API. 

> An array of file path globs to design token files that contain default styles. Style Dictionary uses this as a base collection of design tokens. The tokens found using the "source" attribute will overwrite tokens found using include.

While `source` is required

> An array of file path globs to design token files. Style Dictionary will do a deep merge of all of the token files, allowing you to organize your files however you want.

blackfalcon replied:
Where this really comes in handy is if we were to want to output all the tokens into a single file where we wanted Auro classic and then a theme to override the values as written. If you include all the files into one glob, then there are duplication warnings.

blackfalcon commented on scripts/config-jetstream.json:
```diff
@@ -0,0 +1,21 @@
+{
+  "include": ["./src/themes/jetstream/color/*.json"],
```
```suggestion
```

blackfalcon commented on src/color/alert.json:
```diff
@@ -2,152 +2,53 @@
-          "comment": "Deprecated token",
-          "reference": "n/a",
-          "version": "3.12.0"
+          "deprecated": true
```
None of the newly deprecated tokens have the additional data required to fill out the 'deprecated tokens' page in the doc site. We should have the following added to each deprecated token. 

`reference` - this is a reference to the new token to be used of there is one designated 
`version` - this is used to communicate with wich release these tokens were listed as deprecated

blackfalcon commented on src/color/background.json:
```diff
@@ -1,4 +1,280 @@
 {
+  "background": {
```
It is part of the core Style Dictionary API that when we use colors that we need to start the JSON with a `color` object. Without this we lost core capabilities to transform values as needed.

jordanjones243 replied:
@blackfalcon this seems like it maybe needs to be a further discussion? I am not really sure who named the tokens, but there was definitely a conscious decision to leave the `color` part out of the naming convention.

blackfalcon commented on src/themes/excursion/color/background.json:
```diff
@@ -0,0 +1,79 @@
+{
```
missign `color` object

blackfalcon commented on src/color/brand.json:
```diff
@@ -1,529 +1,1593 @@
 {
+  "classic": {
```
missing `color` object

blackfalcon commented on src/color/container.json:
```diff
@@ -0,0 +1,512 @@
+{
```
missing `color` object

blackfalcon commented on src/color/icon.json:
```diff
@@ -1,148 +1,298 @@
 {
+  "icon": {
```
missing `color` object

jordanjones243 committed:
fix: add color object to new token structure

blackfalcon committed:
refactor: update tokens names

Changes to be committed:
modified:   src/color/background.json
modified:   src/color/base.json
modified:   src/color/container.json
modified:   src/color/text.json

blackfalcon committed:
build: update configs

This update addresses the readability order of the configs to be more
clear when illustrating files that are loaded and which files when
loaded will over-ride the previously loaded files.

Additionally the excursion and jetstream configs were  updated to
eliminate the unnecessary over-ride inclusion.

Changes to be committed:
modified:   package.json
modified:   scripts/config-darkmode.json
modified:   scripts/config-excursion.json
modified:   scripts/config-jetstream.json
modified:   scripts/config.json

blackfalcon commented on src/color/background.json:
```diff
@@ -1,282 +1,280 @@
+      "success": {
         "default": {
-          "value": "{classic.neutral.200.value}",
+          "value": "{color.base.success.100.value}",
```
Token doesn't exist.

blackfalcon commented on src/color/background.json:
```diff
@@ -1,282 +1,280 @@
         },
         "inverse": {
-          "value": "{classic.navy.800.value}",
+          "value": "{color.base.success.500.value}",
```
Token doesn't exist.

blackfalcon commented on src/color/background.json:
```diff
@@ -1,282 +1,280 @@
+      "warning": {
         "default": {
-          "value": "{classic.grey.200.value}",
+          "value": "{color.base.warning.100.value}",
```
Token doesn't exist.

blackfalcon commented on src/color/background.json:
```diff
@@ -1,282 +1,280 @@
         },
         "inverse": {
-          "value": "{classic.navy.700.value}",
+          "value": "{color.base.warning.500.value}",
```
Token doesn't exist.

blackfalcon committed:
refactor: update incorrect refs

Changes to be committed:
modified:   src/color/background.json
modified:   src/color/border.json
modified:   src/color/container.json
modified:   src/color/icon.json
modified:   src/color/text.json

jordanjones243 commented on src/color/background.json:
```diff
@@ -151,15 +151,15 @@
       "error": {
         "default": {
-          "value": "{color.base.error.100.value}",
+          "value": "{color.state.error.100.value}",
```
According to this file, we should be using `base` not `state`:
- https://alaskaair-my.sharepoint.com/:x:/p/june_lee/EeyDi0RTK5xEosX7r0KZ_7MBIayy5sUdJ7WBT4XJZqHUmQ?e=ULdhyD&ovuser=0f44c5d4-42b0-45c2-bf55-d0fea8430d33%2CJordan.Jones4%40alaskaair.com&clickparams=eyJBcHBOYW1lIjoiVGVhbXMtRGVza3RvcCIsIkFwcFZlcnNpb24iOiI1MC8yNDA0MTEyMjMxMyIsIkhhc0ZlZGVyYXRlZFVzZXIiOmZhbHNlfQ%3D%3D

jordanjones243 commented on src/color/background.json:
```diff
@@ -151,15 +151,15 @@
         },
         "inverse": {
-          "value": "{color.base.error.500.value}",
+          "value": "{color.state.error.500.value}",
```
Revert all `state` changes back to `error`

jordanjones243 commented on src/color/base.json:
```diff
@@ -2,150 +2,346 @@
         }
       }
     },
     "state": {
```
According to figma, `state` tokens should be deprecated and there should be new `base` tokens

jordanjones243 commented on src/color/base.json:
```diff
@@ -2,150 +2,346 @@
   "color": {
```
`base-black` and `base-white` tokens are now on the 100s scale
